packages/fxa-settings/src/pages/ResetPassword/ResetPasswordWithRecoveryKeyVerified/container.tsx

```diff
@@ -55,11 +57,17 @@ const ResetPasswordWithRecoveryKeyVerifiedContainer = ({
   const navigateToHint = () => setShowHint(true);
   const navigateNext = async (continueToAccountEvent: () => void) => {
     if (isOAuthIntegration(integration)) {
+      // If we have lost the required bits for OAuth handling, we have to start again.
+      if (!sessionToken || !uid) {
```

Comment:
Alternatively, we could choose to only store the `keyFetchToken` and `unwrapBKey` in the `sensitiveDataClient`, and pass the `uid` and `sessionToken` in location state as these are not as sensitive?

Reply:
@vpomerleau Yeah, I'm game to do this. Wdyt about the `/signin` navigation? I wasn't really sure if this was the right way to solve it. I tried the original bug again and I saw that the STR authenticates, but doesn't have the redirect stored so that we go back to the RP.

Reply:
+1 to this, though instead of location state, I think we always want to just read from local storage / `currentAccount()`. I used to try to optimize against local storage reads but then Dan pointed out just how lightning fast it is. 😅

Reply:
Thanks for the hint!